lua/starfall/libs_cl/render.lua

```diff
@@ -2399,6 +2399,26 @@ function render_library.depthRange(min, max)
 	render.DepthRange(min, max)
 end
 
+--- Creates a new PixVis handle. See render.pixelVisible.
```

Comment:
I think we should use a `SF.ResourceHandler` to handle the PixelVisibleHandles. Each call to pixelVisible will use an unused handle and then at the end of the frame, all the used handles will be returned to the ResourceHandler.

Reply:
Then pixelVisible would just be `render_library.pixelVisible(position,radius)`

Reply:
This seems more than sensible (I understand what you are asking of the code), but I'm not familiar with how SF.ResourceHandler is meant to be used? I examined the code for it and read one instance of it earlier in render.lua, but I honestly just don't have the right grasp to where I could competently put something together with it.

Reply:
```lua
local pixhandle_bank = SF.ResourceHandler(cv_max_pixelhandles:GetInt(),
	function()
		return util.GetPixelVisibleHandle()
	end, nil, function() return 1 end
)
```

```lua
local usedPixelVis = {}
local function freePixelVis()
    for _, v in ipairs(usedPixelVis) do
        pixhandle_bank:free(instance.player, v)
    end
end
```

Then in pixelVisible() you just use
```lua
local PixVis = pixhandle_bank:use(instance.player, 1)
if #usedPixelVis==0 then
    hook.Add("SomePostRenderHook","SF_FreePixelVis",freePixelVis)
end
usedPixelVis[#usedPixelVis + 1] = PixVis
return util.PixelVisible(vunwrap(position),radius,PixVis)
```

Reply:
I spent some time last night trying to get these to work, but I couldn't. I'm not certain why, but even when the pixelvis_handle_t was being supplied to the function, the result is garbage (it's always zero, even when I'm certain it should not be. I am using the same basic setup to test this as I was with the original functions, so I know when the result differs from what I should expect.)
Not only that, but after exactly 4 executions, nil is returned by pixhandle_bank, and the function *never* returns a pixelvis_handle_t after that.

I tried to investigate both issues, but I really can't say I was able to understand what was going on -- especially with the first problem. The documentation util.GetPixelVisibleHandle has is lacking.

Reply:
The latter issue I'm certain is more than solvable -- I had a lot of confusion trying to understand the problem before, and by the time I did understand it I needed to sleep. But the former boggles me. Out of curiosity, I tried doing `return util.PixelVisible(vunwrap(position),radius,util.GetPixelVisibleHandle())` just to see if that would also give the garbage result, and it does.

Reply:
@Advers maybe the handle has to be created before the scene is rendered. Maybe if we initialize the resource handler with handles, it won't have issues anymore. I can try it out if you can go ahead and push your progress.

Reply:
It might also depend on what hook you're calling util.PixelVisible from.